engine/engine.go

```diff
@@ -974,13 +998,36 @@ func (e *Engine) DbPTUnref(db string, ptId uint32) {
 	e.mu.RUnlock()
 }
 
-func (e *Engine) GetShard(db string, ptId uint32, shardID uint64) Shard {
+func (e *Engine) GetShard(db string, ptId uint32, shardID uint64) (Shard, error) {
 	pt := e.getDBPTInfo(db, ptId)
 	if pt == nil {
-		return nil
+		return nil, nil
+	}
+
+	sh := pt.Shard(shardID)
+	if sh == nil {
```

Comment:
Is there any case skip shard not found error? If not, it is more appropriate to return error ShardNotFound.

Reply:
The logic here is consistent with `if pt == nil {` above.

https://github.com/openGemini/openGemini/blob/4f3a950962deee00534c9eae461e9464ed7ddbaf/engine/engine.go#L977-L981

Reply:
Do not introduce new logic, worrying about affecting existing processing logic.